streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfiguration.java

```diff
@@ -108,106 +110,20 @@ public RebalanceProtocol rebalanceProtocol() {
                 case UPGRADE_FROM_21:
                 case UPGRADE_FROM_22:
                 case UPGRADE_FROM_23:
-                    // ATTENTION: The following log messages is used for verification in system test
-                    // streams/streams_cooperative_rebalance_upgrade_test.py::StreamsCooperativeRebalanceUpgradeTest.test_upgrade_to_cooperative_rebalance
-                    // If you change it, please do also change the system test accordingly and
-                    // verify whether the test passes.
-                    log.info("Eager rebalancing protocol is enabled now for upgrade from {}.x", upgradeFrom);
-                    log.warn("The eager rebalancing protocol is deprecated and will stop being supported in a future release." +
-                        " Please be prepared to remove the 'upgrade.from' config soon.");
-                    return RebalanceProtocol.EAGER;
-                case UPGRADE_FROM_24:
-                case UPGRADE_FROM_25:
-                case UPGRADE_FROM_26:
-                case UPGRADE_FROM_27:
-                case UPGRADE_FROM_28:
-                case UPGRADE_FROM_30:
-                case UPGRADE_FROM_31:
-                case UPGRADE_FROM_32:
-                case UPGRADE_FROM_33:
-                case UPGRADE_FROM_34:
-                case UPGRADE_FROM_35:
-                case UPGRADE_FROM_36:
-                case UPGRADE_FROM_37:
-                case UPGRADE_FROM_38:
-                case UPGRADE_FROM_39:
-                    // we need to add new version when new "upgrade.from" values become available
+                    final String errMsg = String.format(
+                        "The eager rebalancing protocol is no longer supported in 4.0 which means live upgrades from 2.3 or below are not possible."
+                            + " Please see the Streams upgrade guide for the bridge releases and recommended upgrade path. Got upgrade.from='%s'", upgradeFrom);
+                    log.error(errMsg);
```

Comment:
Why do `log.error` here? When we throw the `ConfigException` it will be logged later a second time.

Reply:
[see above](https://github.com/apache/kafka/pull/18988#discussion_r1964707959)

Reply:
I guess in this case it's not as essential to know where the error came from since it's a simple one, but I still prefer more logging over less logging. Sometimes exception messages just get eaten (for a variety of different reasons in my experience)

Reply:
SG.